pkg/libartifact/store/store.go

```diff
@@ -335,22 +331,10 @@ func getManifests(ctx context.Context, imgSrc types.ImageSource, startingDigest
 		if err != nil {
 			return nil, err
 		}
-		manifests = append(manifests, *mani)
-		return manifests, nil
-	}
-	// We are dealing with an oci index list
-	maniList, err := manifest.OCI1IndexFromManifest(b)
-	if err != nil {
-		return nil, err
-	}
-	for _, m := range maniList.Manifests {
-		iterManifests, err := getManifests(ctx, imgSrc, &m.Digest)
-		if err != nil {
-			return nil, err
-		}
-		manifests = append(manifests, iterManifests...)
+		return mani, nil
 	}
-	return manifests, nil
+	// We are dealing with an oci index list which should not be the case in our own store.
+	return nil, fmt.Errorf("manifest %q is index list", imgSrc.Reference().StringWithinTransport())
```

Comment:
Non-blocking: Aesthetically I’d prefer this to happen as an early exit, so that the straight line is the base case.

Also, the “these are the keepers” comment (and the comment above that?) could be updated.

Reply:
done